### PR TITLE
SDK - Testing - Run some unit-tests in a more correct way

### PR DESCRIPTION
**Pull request by Ark-kun**
Replaced `@unittest.expectedFailure` with `with self.assertRaises(...):`.
Replaced `assert` with `self.assertEqual(...)`.
Stopped producing the stray "comp.yaml" file.
Enabled the test_load_component_from_url test.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2036)

### CODE REVIEW

File: sdk/python/tests/components/test_components.py [hunk 1]
```diff
@@ -62,26 +62,25 @@ def test_load_component_from_zipped_yaml_file(self):
         component_path = _test_data_dir.joinpath('python_add.component.zip')
         self._test_load_component_from_file(str(component_path))
 
-    @unittest.skip
-    @unittest.expectedFailure #The repo is non-public and will change soon. TODO: Update the URL and enable the test once we move to a public repo
+    #The repo is non-public and will change soon. TODO: Update the URL and enable the test once we move to a public repo
```
Comment (numerology) [inline sdk/python/tests/components/test_components.py]: Not sure what this comment mean. Is it a stale one?

File: sdk/python/tests/components/test_python_op.py [hunk 2]
```diff
@@ -348,7 +348,7 @@ def assert_is_none(a, b, arg=None) -> int:
             return 1
 
         func = assert_is_none
-        op = comp.func_to_container_op(func, output_component_file='comp.yaml')
+        op = comp.func_to_container_op(func)
```
Comment (numerology) [inline sdk/python/tests/components/test_python_op.py]: I think removing this param will make the coverage of the unittest a bit of smaller.
If output_component_file is not specified, component_spec won't be translated into a yaml dump. WDYT?

Comment (Ark-kun) [reply sdk/python/tests/components/test_python_op.py]: Thanks. Did not think about that. That piece of code would only be a couple of lines, but we might want to test it.

File: sdk/python/tests/components/test_components.py (hunk 1 above)
Comment (Ark-kun) [reply sdk/python/tests/components/test_components.py]: It's stale. I'll remove it.

Comment (Ark-kun) [reply sdk/python/tests/components/test_components.py]: I've removed that line.

File: sdk/python/tests/components/test_python_op.py (hunk 2 above)
Comment (Ark-kun) [reply sdk/python/tests/components/test_python_op.py]: I've added two new test that test the writing components to file.

**Comment by chases2** [comment]
/hold

There's an issue with getting this to merge. I'm looking into this now.
`failed merging [2036]: [PR is unmergable. Do the Tide merge requirements match the GitHub settings for the repo? Required status check "Travis CI - Pull Request" is queued.]`

**Comment by Ark-kun** [comment]
@chases2 Is it possible to fix the issue on our side or the only way is to stop using Travis?

**Comment by Ark-kun** [comment]
@numerology Can you please re-LGTM - Travis checks have malfunctioned again.

**Pull request merged**